### ISSUE TITLE
requestPermissionToSendNotifications() never finishes

### ISSUE THREAD

**Issue by myrkgard** [rafaelsetragni/awesome_notifications#80]
Hi,
in ```example/lib/pages/notification_examples_page.dart```, starting from line 219, there is:

```
onOkButtonPressed: () async {
  Navigator.of(context).pop();
  await AwesomeNotifications().requestPermissionToSendNotifications();
  notificationsAllowed = await AwesomeNotifications().isNotificationAllowed(); // NEVER REACHED
  setState(() {
    notificationsAllowed = notificationsAllowed;
  });
},
```

Here, ```await AwesomeNotifications().requestPermissionToSendNotifications();``` never finishes, thus the code following it is never executed. The notification dialog of the OS is shown, though. Observed on Android 8.1 and 11.

**Comment by rafaelsetragni** [comment]
Ive made a modification based on [this topic ](https://stackoverflow.com/questions/32366649/any-way-to-link-to-the-android-notification-settings-for-my-app).

Try to add in your local files the code bellow, inside the file AwesomeNotificationsPlugin.java(line 625):

```Java

    private void channelRequestNotification(MethodCall call, Result result){

        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            
            final Intent intent = new Intent();

            intent.setAction(Settings.ACTION_APP_NOTIFICATION_SETTINGS);

            //for Android 5-7
            intent.putExtra("app_package", applicationContext.getPackageName());
            intent.putExtra("app_uid", applicationContext.getApplicationInfo().uid);

            // for Android 8 and above
            intent.putExtra("android.provider.extra.APP_PACKAGE", applicationContext.getPackageName());

            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);

            applicationContext.startActivity(intent);
            result.success(true);
        }
        else {
            channelIsNotificationAllowed(call, result);
        }
    }
```

**Comment by myrkgard** [comment]
Thank you! Works now. Except the example doesn't work as intended: As ```requestPermissionToSendNotifications()``` now finishes immediately, the code below is executed before the user can make any changes in the called OS dialog. But I don't think there is anything one can do about it (?). So I suggest to adjust the example to the new realities.

**Comment by rafaelsetragni** [comment]
This may be because you are trying to send the notification right after the notification request. As far as i know, is not possible to detect when the user activates the notification and come back to your app.

**Comment by myrkgard** [comment]
Yes, I don't think it's possible to detect it either. What I meant by "example doesn't work as intended": The green / red indicator on the example page is not guarantied to correctly reflect if notifications are allowed or not at the moment. Maybe, instead of the green / red indicator, the example should have a button "Check if allowed" that calls ```isNotificationAllowed()``` and shows the result in a Toast.

**Comment by rafaelsetragni** [comment]
Yes, its a refresh screen problem. I don't know yet how to fix it.

I gonna release a new version soon to fix the problem that you found. Thank you very much for your help!
